confgenerator/logging.go

```diff
@@ -48,12 +48,17 @@ func setLogNameComponents(ctx context.Context, tag, logName, receiverType string
 }
 
 // stackdriverOutputComponent generates a component that outputs logs matching the regex `match` using `userAgent`.
-func stackdriverOutputComponent(ctx context.Context, match, userAgent string, storageLimitSize string) fluentbit.Component {
+func stackdriverOutputComponent(ctx context.Context, match, userAgent string, storageLimitSize string) (fluentbit.Component, error) {
+	r, err := platform.FromContext(ctx).GetResource()
```

Comment:
The old code explicitly didn't do resource detection here because it didn't want to block on it; why is this change necessary?

Reply:
Line 83 in the old code did the resource override